app/controllers/admin/enterprises_controller.rb

```diff
@@ -32,6 +32,12 @@ def index
       end
     end
 
+    def edit
+      @object = Enterprise.where(permalink: params[:id]).
+        includes(users: [:ship_address, :bill_address]).first
+      super
+    end
+
```

Comment:
the load_resource method in the resource controller will still execute with the N+1, right?

Reply:
No, `load_resource` starts with:
```ruby
def load_resource
  if member_action?
    @object ||= load_resource_instance

...
```
and `bullet` confirmed the N+1 was gone.

Reply:
awesome!